File: backend/shop-floor-assistance-backend/src/main/java/com/shopfloor/backend/api/transferobjects/operators/OperatorExecutionTO.java
package com.shopfloor.backend.api.transferobjects.operators;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;

/**
 * Transfer object for operator execution.
 * Contains details about the execution, including its start and finish times, whether it was aborted, and the IDs of the users who started and finished it.
 * @author David Todorov (https://github.com/david-todorov)
 */
@Getter
@Setter
public class OperatorExecutionTO {
    /**
     * The unique identifier for the operator execution.
     */
    private Long id;

    /**
     * The date and time when the execution started.
     */
    private Date startedAt;

    /**
     * The date and time when the execution finished.
     */
    private Date finishedAt;

    /**
     * Indicates whether the execution was aborted.
     */
    private Boolean aborted;

    /**
     * The ID of the user who started the execution.
     */
    private Long startedBy;

    /**
     * The ID of the user who finished the execution.
     */
    private Long finishedBy;
}

File: frontend/shop-floor-assistance-frontend/src/app/shared/types/OperatorExecutionTO.ts
export type OperatorExecutionTO = {
    id: number;                // Unique execution ID
    started_at: string;        // Start timestamp
    finished_at?: string | null; // Finish timestamp (nullable)
    started_by: number;        // ID of the user who started
    finished_by?: number | null; // ID of the user who finished (nullable)
    order_id: number;          // Associated order ID
    aborted: boolean;          // Aborted status
};
